### ISSUE TITLE
door being open/closed gets out of sync with reality

### ISSUE THREAD

**Issue by jhaar** [sam43434/remootio#19]
Hi there

I've been using this custom addon for about a year, and am the sort of person who continually updates HA with the official releases. I run it via docker. Several months ago (after working fine until then) I noticed the garage door state would get out of sync with reality. Door tagged as "open" when it's "closed" kinda thing. Usually restarting HA would fix that. But as the monthly releases continued, it's got less and less reliable - these days even restarting HA doesn't seem to fix it. Once it's out of sync - it stays like that - until it gets out of sync again - and so ends up working again! 

BTW I do have the official Remootio app on my Android and it's always worked - so this isn't anything to do with the hardware/magnet/etc. We also still use the "real" door opener successfully - and maybe it's the reason this gets out of whack? i.e. HA closes the garage, real remote opens it and HA only knows it's last command state? I dunno - just guessing.

I know there's not much to go on there  - I am happy to generate diagnostics if anyone can help - and tell me what to do?

Thanks

Jason

**Comment by alpha520098** [comment]
have you tried a fresh install?

**Comment by jhaar** [comment]
Good idea. I downloaded it via "git clone https://github.com/sam43434/remootio.git" and diff'ed it against the running version. No changes (not quite true, I just added the https://github.com/sam43434/remootio/commit/0f1b0c1e639639a9801f6d0d5c859de55e9e1d56 changes last week as it had entirely broken - so those changes are expected).

So that doesn't appear to be the problem.

**Comment by alpha520098** [comment]
Instructions below to fix remootio to work with HA 7.3

Go to github and fork the aioremootio repo
https://github.com/ivgg-me/aioremootio

edit the requirements.txt file in your own aioremootio fork and change to below versions
aiohttp==3.8.5
pycryptodome==3.17
async-class==0.5.0
voluptuous==0.13.1

Go to your HA /config/custom_components/remootio/

edit manifest.json file and change the line with requirements to look like the below
"requirements": ["aioremootio @ git+https://github.com/GITHUBUSERID/aioremootio.git"],
but change GITHUBUSERID to yours

Don't redownload remootio HACS Otherwise you will overwrite your changes done in step 3

Restart HA and you should be good to go.

theres is a thread with 54 comments - i can assure you it works even with the latest update

**Comment by jhaar** [comment]
I didn't say it didn't work? You need that patch to make remootio work at all. I'm saying my problem with the up/down confusion predates that drama and is the reason my HA remootio directory is slightly different from the git repo

**Comment by osterman** [comment]
I have also encountered this lately and also noticed rebooting HA seems to resolve itself, for a while. I applied the manual patches and to get it working on the latest HA, but it doesn't address the issue with it getting out of sync.
